azurerm/internal/services/hpccache/hpc_cache_resource.go

```diff
@@ -454,3 +483,50 @@ func flattenStorageCacheNfsAccessRules(input *[]storagecache.NfsAccessRule) ([]i
 
 	return rules, nil
 }
+
+func expandStorageCacheNetworkSettings(d *schema.ResourceData) *storagecache.CacheNetworkSettings {
+	out := &storagecache.CacheNetworkSettings{
+		Mtu:       utils.Int32(int32(d.Get("mtu").(int))),
+		NtpServer: utils.String(d.Get("ntp_server").(string)),
+	}
+
+	if dnsSetting, ok := d.GetOk("dns_setting"); ok {
+		dnsSetting := dnsSetting.([]interface{})[0].(map[string]interface{})
+		out.DNSServers = utils.ExpandStringSlice(dnsSetting["servers"].([]interface{}))
+		searchDomain := dnsSetting["search_domain"].(string)
+		if searchDomain != "" {
+			out.DNSSearchDomain = &searchDomain
+		}
+	}
+	return out
+}
+
+func flattenStorageCacheNetworkSettings(settings *storagecache.CacheNetworkSettings, d *schema.ResourceData) error {
+	if settings == nil {
+		return nil
+	}
+
+	d.Set("mtu", settings.Mtu)
+	d.Set("ntp_server", settings.NtpServer)
+
+	if settings.DNSServers != nil {
+		dnsServers := utils.FlattenStringSlice(settings.DNSServers)
+
+		searchDomain := ""
+		if settings.DNSSearchDomain != nil {
+			searchDomain = *settings.DNSSearchDomain
+		}
+
+		dnsSetting := []interface{}{
+			map[string]interface{}{
+				"servers":       dnsServers,
+				"search_domain": searchDomain,
+			},
+		}
+
+		if err := d.Set("dns_setting", dnsSetting); err != nil {
```

Comment:
this shouldn't be settings things within this function, can we return an empty array (and above where this is nil - to track changes for the nil case) and have this in the read?